File: C++/longest_harmonious_subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        if(nums.empty()) return 0;
        
        sort(nums.begin(), nums.end());
        int result = 0, count = 1, prev = 0;
        for(int i = 1; i < nums.size(); i++) {
            if(nums[i] == nums[i-1]) count ++;
            else {
                if(prev && prev + count > result) 
                    result = prev + count;
                prev = (nums[i] - nums[i-1] == 1) ? count : 0;
                count = 1;
            }
        }
        
        return max(result, prev ? count + prev : 0);
    }
};

File: Python3/longest_harmonious_subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        result = 0
        
        if not nums: return result
        
        dict = {}
        for num in nums:
            if num not in dict:
                dict[num] = 1
                continue
            dict[num] += 1
        
        keys = sorted(list(dict.keys()))
        for i in range(len(keys) - 1):
            if keys[i] + 1 == keys[i+1]:
                sum2 = dict[keys[i]] + dict[keys[i+1]]
                if sum2 > result: result = sum2
        
        return result
